frontend/src/app/main/ui/workspace/sidebar/options/menus/component.cljs

```diff
@@ -449,47 +450,44 @@
-             (let [target-props (-> (:variant-properties component)
-                                    (update pos assoc :value val))
-                   valid-comps  (->> variant-comps
-                                     (remove #(= (:id %) component-id))
-                                     (filter #(= (dm/get-in % [:variant-properties pos :value]) val))
-                                     (reverse))
-                   nearest-comp (apply min-key #(ctv/distance target-props (:variant-properties %)) valid-comps)
-                   parents (cfh/get-parents-with-self page-objects (:parent-id shape))
-                   children (cfh/get-children-with-self objects (:main-instance-id nearest-comp))
-                   comps-nesting-loop? (seq? (cfh/components-nesting-loop? children parents))]
-
-               (when nearest-comp
-                 (if comps-nesting-loop?
-                   (do
-                     (st/emit! (ntf/error (tr "workspace.component.swap.loop-error")))
-                     (reset! key* (uuid/next)))
-                   (st/emit! (dwl/component-swap shape (:component-file shape) (:id nearest-comp) true))))))))]
+           (if (= val mixed-label)
+             (reset! key* (uuid/next))
+             (let [error-msg (if (> (count shapes) 1)
+                               (tr "workspace.component.switch.loop-error-multi")
+                               (tr "workspace.component.swap.loop-error"))
+                   mdata     {:on-error #(do
+                                           (st/emit! (ntf/error error-msg))
+                                           (reset! key* (uuid/next)))}
+                   params    {:shapes shapes :pos pos :val val}]
+               (st/emit! (dwv/variants-switch (with-meta params mdata)))))))]
 
     [:*
      [:div {:class (stl/css :variant-property-list)}
-      (for [[pos prop] (map vector (range) properties)]
-        [:div {:key (str (:id shape) pos)
-               :class (stl/css :variant-property-container)}
-
-         [:div {:class (stl/css :variant-property-name-wrapper)
-                :title (:name prop)}
-          [:div {:class (stl/css :variant-property-name)}
-           (:name prop)]]
-
-         [:div {:class (stl/css :variant-property-value-wrapper)}
-          [:> select* {:default-selected (:value prop)
-                       :options (get-options (:name prop))
-                       :empty-to-end true
-                       :on-change (partial switch-component pos)
-                       :key (str (:value prop) "-" key)}]]])]
+      (for [[pos prop] (map vector (range) props-first)]
+        (let [mixed-value? (not-every? #(= (:value prop) (:value (nth % pos))) properties)
+              options (cond-> (get-options (:name prop))
+                        mixed-value?
+                        (conj {:id mixed-label, :label mixed-label :dimmed true}))]
+          [:div {:key (str (:id shape) pos mixed-value?)
```

Comment:
The key should be unique on the iteration, i think with `pos` is more than enough for the key, so you can remove the string concat operation and replace it with just using `pos`